### PR TITLE
ScrollSpy: make the threshold option configurable

### PR DESCRIPTION
**Pull request by GeoSot**
ref #36431

### CODE REVIEW

File: js/tests/unit/scrollspy.spec.js [hunk 1]
```diff
@@ -126,6 +126,49 @@ describe('ScrollSpy', () => {
...
+        '<div id="content" data-bs-threshold="0,0.2,1">',
+        '  <div id="one-link">test</div>',
+        '</div>'
+      ].join('')
+
+      const scrollSpy = new ScrollSpy('#content', {
+        target: '#navigation'
+      })
+
+      const expectToBeCloseToArray = (actual, expected) => {
```
Comment (julien-deramond) [inline js/tests/unit/scrollspy.spec.js:161]:
```suggestion
      // See https://stackoverflow.com/a/45592926
      const expectToBeCloseToArray = (actual, expected) => {
```

File: js/tests/unit/scrollspy.spec.js [hunk 2]
```diff
@@ -126,6 +126,49 @@ describe('ScrollSpy', () => {
...
+
+      const scrollSpy = new ScrollSpy('#content', {
+        target: '#navigation',
+        threshold: [1]
+      })
+
+      expect(scrollSpy._observer.thresholds).toEqual([1])
+    })
+
+    it('should respect threshold option mark-up', () => {
```
Comment (julien-deramond) [inline js/tests/unit/scrollspy.spec.js:147]:
```suggestion
    it('should respect threshold option markup', () => {
```

File: js/src/scrollspy.js [hunk 3]
```diff
@@ -110,6 +112,13 @@ class ScrollSpy extends BaseComponent {
     // TODO: on v6 target should be given explicitly & remove the {target: 'ss-target'} case
     config.target = getElement(config.target) || document.body
 
+    // TODO: v6 Only for backwards compatibility reasons. Use rootMargin only
+    config.rootMargin = config.offset ? `${config.offset}px 0px -30%` : config.rootMargin
+
+    if (typeof config.threshold === 'string') {
```
Comment (julien-deramond) [inline js/src/scrollspy.js:118]: Might be a dumb question but do we really need to handle string param?
In the unit tests if I do `'<div id="content" data-bs-threshold="[0,0.2,1]">',` it seems to work as well (haven't tried it in a real example).

**Review by julien-deramond — CHANGES_REQUESTED** [review]
I haven't tested how it works in details but the mechanism to provide the `threshold` param seems to work pretty well.
Just few proposals of improvement and a question on my side.

File: js/src/scrollspy.js (hunk 3 above)
Comment (GeoSot) [reply js/src/scrollspy.js:118]: If it was on me and only, I wouldn't do it
 However, all our options are being parsed as strings too, so I am pretty sure that on the launch  day, we will face a new issue

Comment (julien-deramond) [reply js/src/scrollspy.js:118]: Oh I see. So let's keep things coherent, you're right :)

**Review by julien-deramond — APPROVED** [review]
LGTM! Don't know if anyone want to double check this one.

**Pull request merged**